src/node/internal/process.ts

```diff
@@ -20,13 +20,90 @@ export function nextTick(cb: Function, ...args: unknown[]) {
   });
 }
 
-// Note that there is no process-level environment in workers so the process.env
-// object is initially empty. This is different from Node.js where process.env
-// picks up values from the operating system environment. The configured bindings
-// for the worker are accessible from the env argument passed into the fetch
-// handler and have no impact here.
+// Decide if a value can round-trip to JSON without losing any information.
+function isJsonSerializable(value: any): boolean {
+  switch (typeof value) {
+    case "boolean":
+    case "number":
+    case "string":
+      return true;
 
-export const env = new Proxy(utilImpl.getEnvObject(), {
+    case "object": {
+      if (value === null) {
+        return true;
+      }
+
+      // We only consider objects to be serializable if they are plain objects or plain arrays.
+      // Any subclass is not serializable because serializing it would be lossy.
+      //
+      // (Practically speaking, we don't want, say, a KV namespace to show up in process.env as
+      // "{}" -- this would be weird.)
+      switch (Object.getPrototypeOf(value)) {
+        case Object.prototype:
+          for (let prop of Object.values(value)) {
+            if (!isJsonSerializable(prop)) return false;
+          }
+          return true;
+        case Array.prototype:
+          for (let elem of value) {
+            if (!isJsonSerializable(elem)) return false;
+          }
+          return true;
```

Comment:
Nit: This can be simplified with an array.prototype.some() call

Reply:
Matter of opinion but honestly I find the loop easier to read.